### ISSUE TITLE
[MU4 Issue] stem direction of notes on middle staff line seems unpredictable

### ISSUE THREAD

**Issue by wizofaus** [musescore/MuseScore#11498]
**Describe the bug**
While entering notes, the stem direction of any unbeamed notes on the middle staff line (e.g. B on the treble clef) seems unpredictable, and often isn't restored after undo-ing operations.  Compare v3 where it's always down.

**To Reproduce**
1. Create a new score, default treble clef, 4/4
2. Enter notes G A B  (all 1/4 notes)
3. Note that B stem is UP (vs v3, where it's down)
4. Enter note C - B stem is unchanged, then D (new measure) - B stem flips down on entering this
5. Undo entering the D. B stem is still down.
6. Arrow left so C is selected, then hit C again - notes unchanged but B stem flips back up again
7. Undo this "non-edit" - this time B stem restores to down position.
8. Use the mouse to drag the C up and down - the B stem flips back and forth somewhat unexpectedly
9. Undo the drag action, the B stem is not restored
10. In some cases just changing what's currently selected causes the stem direction to flip. Hard to come up with a reliable way to reproduce.

**Expected behavior**
TBH I think v3's behaviour (B stem is always down) made more sense, even if it doesn't follow engraving best practices...but at any rate the behaviour should be more predictable than it is, especially on undo.  I'm not sure why there should be a case where adding a note in the next bar/measure would cause automatical stem flips.

**Screenshots**

**Platform information**
 - OS: [e.g. macOS, Windows, Linux]

**Additional context**

**Comment by oktophonie** [comment]
Behaviour should indeed be like in MU3 - there was a brief flirtation with having this be context-dependent some time back but this was removed in [#9969](https://github.com/musescore/MuseScore/pull/9969); perhaps something was overlooked there?

**Comment by Tantacrul** [comment]
I'm aware that logic has been created to make the stem direction context aware and not just a hard rule. I understand this was done to minimise beam heights.

Closing for now, unless @oktophonie recognises a bug he wants fixed.

**Comment by Tantacrul** [comment]
Assigned the engraving crew. Please feel free to reprioritse / reassign, etc.

**Comment by wizofaus** [comment]
Most likely explanation is that my "preferStemDirectionMatchContext" setting was still true because that was the default at some point the past, or that's what it is my template files?
It doesn't appear there's any UI for changing preferStemDirectionMatchContext and the hard-coded factory default is definitely false, so I'm not 100% sure where my "true" setting is coming from.

**Comment by cbjeukendrup** [comment]
Indeed, the template files: https://github.com/musescore/MuseScore/blob/5cc7617665213ffcc36bef78f6ec236e9f029654/share/templates/01-General/01-Treble_Clef/score_style.mss#L132
(Well, I checked only one, but I fear that it's not the only one)

Good news is that it is pretty trivial to fix, in that case.

**Comment by wizofaus** [comment]
Ah, .mss, I didn't think to search for that!  But yeah, fixing that will solve the issue for now, but if in future there is a plan to reintroduce the functionality I'd say it needs some more work, certainly around undo and the fact that just changing what's selected can cause stems to flip.

**Comment by oktophonie** [comment]
Yes, `preferStemDirectionMatchContext` should be false in all templates, indeed all files. My preference would be to remove this option and all associated functionality completely, anyway.

**Comment by wizofaus** [comment]
I'd observed stems changing randomly too, even when just changing what notes are selected, so it's possible there still is another bug here though I couldn't reproduce it.

**Comment by wizofaus** [comment]
I can easily put up a PR that just sets preferStemDirectionMatchContext to false in the templates at least (or remove it from the templates if that works), which will solve the most obvious/likely cases users will face.  Or is it better done via a dev setting (assuming we still want to leave the code in place at all)?
@oktophonie

**Comment by oktophonie** [comment]
My preference is to remove all of this code entirely! But removed from all the templates at the very least.
